### PR TITLE
[geometry] Define serialize/deserialize steps

### LINKED ISSUE #121: [geometry] Update for structured cloning changes in HTML
https://drafts.fxtf.org/geometry/#cloning should be formalized and reformulated in terms of HTML's (recently-introduced, in https://github.com/whatwg/html/commit/97d644c97335956610a31e8ad98d1a388c063e840) serialization/deserialization steps.

I can work on a PR for this shortly.

### PR DESCRIPTION
**Pull request by zcorpan**
Fixes #121.

---

@domenic r?

### CODE REVIEW

**Comment by zcorpan** [comment]
or @annevk

File: geometry/Overview.bs [hunk 1]
```diff
@@ -1713,21 +1720,167 @@ user agents.
...
+
+ <li>Set |value|'s <a for=point>y coordinate</a> to |serialized|.\[[Y]].
+
+ <li>Set |value|'s <a for=point>z coordinate</a> to |serialized|.\[[Z]].
+
+ <li>Set |value|'s <a for=point>w perspective</a> to |serialized|.\[[W]].
+</ol>
+
+<p>The <a>serialization steps</a> for {{DOMRectReadOnly}} and {{DOMRect}}, given |value| and
+|serialized|, are:
```
Comment (domenic) [inline geometry/Overview.bs]: These don't serialize the origin, which implies to me origin might not be a peer of x/y/width/height, and as such might not belong in "Rectangles have the following properties:"

File: geometry/Overview.bs [hunk 2]
```diff
@@ -1713,21 +1720,167 @@ user agents.
...
+ <li>Set |value|'s <a for=rectangle>width dimension</a> to |serialized|.\[[Width]].
+
+ <li>Set |value|'s <a for=rectangle>height dimension</a> to |serialized|.\[[Height]].
+</ol>
+
+<p>The <a>serialization steps</a> for {{DOMQuad}}, given |value| and |serialized|, are:
+
+<ol>
+ <li><p>Set |serialized|.\[[P1]] to the <a>sub-serialization</a> of |value|'s <a
+ for=quadrilateral>point 1</a>.
```
Comment (domenic) [inline geometry/Overview.bs]: Woohoo, sub-serialization proves its use!

File: geometry/Overview.bs [hunk 3]
```diff
@@ -1713,21 +1720,167 @@ user agents.
 
 <h2 id="cloning">Cloning</h2>
```
Comment (domenic) [inline geometry/Overview.bs]: Maybe rename to Serialization (or Structured serialization) and use oldids="cloning"?

**Comment by zcorpan** [comment]
Bugs filed about implementing this:
https://bugzilla.mozilla.org/show_bug.cgi?id=1364049
https://bugs.chromium.org/p/chromium/issues/detail?id=721331

File: geometry/Overview.bs (hunk 3 above)
Comment (zcorpan) [inline geometry/Overview.bs]: Done.

File: geometry/Overview.bs (hunk 1 above)
Comment (zcorpan) [inline geometry/Overview.bs]: Filed https://github.com/w3c/fxtf-drafts/issues/150 to follow up on this, thanks!

**Pull request merged**